api/src/components/bookmark/controller.ts

```diff
@@ -1,134 +1,130 @@
-            return response.json(404, {
-                message: 'You cannot bookmark a draft publication.'
-            });
-        }
+        // Generic checks
 
-        const bookmark = await bookmarkService.get(event.pathParameters.id, event.user.id);
+        // Check that the bookmark exists
+        const bookmark = await bookmarkService.get(event.pathParameters.id);
 
-        // check that the user has created the bookmark/ hasn't already deleted it
         if (!bookmark) {
             return response.json(404, {
-                message: 'You have already deleted the bookmark for this publication.'
+                message: 'This bookmark does not exist.'
             });
         }
 
-        await bookmarkService.remove(bookmark.id);
-
-        return response.json(200, { message: 'Bookmark deleted from this publication.' });
-    } catch (err) {
-        return response.json(500, { message: 'Unknown server error.' });
-    }
-};
-
-export const get = async (
-    event: I.AuthenticatedAPIRequest<undefined, undefined, I.GetBookmarkPathParams>
-): Promise<I.JSONResponse> => {
-    try {
-        const publication = await publicationService.get(event.pathParameters.id);
-
-        //check that the publication exists
-        if (!publication) {
-            return response.json(404, {
-                message: 'This publication does not exist.'
-            });
-        }
-
-        //check that the publication is live
-        if (publication.currentStatus === 'DRAFT') {
-            return response.json(404, {
-                message: 'This publication is in a draft state, so no bookmark exists.'
+        // Check that the user owns the bookmark
+        if (bookmark.userId !== event.user.id) {
+            return response.json(403, {
+                message: 'You do not own this bookmark.'
             });
         }
 
-        const bookmark = await bookmarkService.get(event.pathParameters.id, event.user.id);
-
-        if (!bookmark) {
-            return response.json(200, false);
-        }
+        // Delete bookmark
+        await bookmarkService.deleteBookmark(bookmark.id);
 
-        return response.json(200, bookmark);
+        return response.json(200, { message: 'Bookmark deleted successfully.' });
     } catch (err) {
         return response.json(500, { message: 'Unknown server error.' });
     }
 };
 
 export const getAll = async (
-    event: I.AuthenticatedAPIRequest<undefined, undefined, I.GetAllBookmarkPathParams>
+    event: I.AuthenticatedAPIRequest<undefined, I.GetAllBookmarksQueryStringParameters>
 ): Promise<I.JSONResponse> => {
     try {
-        const bookmarks = await bookmarkService.getAll(event.user.id);
+        const typeFilter: I.BookmarkType = event.queryStringParameters.type;
+        const entityFilter = event.queryStringParameters.entityId || undefined;
+        const bookmarks = await bookmarkService.getMany(event.user.id, typeFilter, entityFilter);
 
         if (!bookmarks) {
```

Comment:
This if statement will never execute since 'bookmarkService.getMany' will always return an array, even if it returns an empty array, !bookmarks will always be false. I think we can safely remove this check and simply return the empty array to the client if that's the case.

Reply:
Well spotted, thanks